staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -146,32 +143,39 @@ func (a *AdmissionOptions) Validate() []error {
 	return errs
 }
 
-// enabledPluginNames makes use of RecommendedPluginOrder and DefaultOffPlugins fields
+// enabledPluginNames makes use of RecommendedPluginOrder, DefaultEnabledPlugins and PluginNames fields
 // to prepare a list of plugin names that are enabled.
-//
-// TODO(p0lyn0mial): In the end we will introduce two new flags:
-// --disable-admission-plugin this would be a list of admission plugins that a cluster-admin wants to explicitly disable.
-// --enable-admission-plugin  this would be a list of admission plugins that a cluster-admin wants to explicitly enable.
-// both flags are going to be handled by this method
 func (a *AdmissionOptions) enabledPluginNames() []string {
-	//TODO(p0lyn0mial): first subtract plugins that a user wants to explicitly enable from allOffPlugins (DefaultOffPlugins)
-	//TODO(p0lyn0miial): then add/append plugins that a user wants to explicitly disable to allOffPlugins
-	//TODO(p0lyn0mial): so that --off=three --on=one,three default-off=one,two results in  "one" being enabled.
-	allOffPlugins := a.DefaultOffPlugins
-	onlyEnabledPluginNames := []string{}
-	for _, pluginName := range a.RecommendedPluginOrder {
-		disablePlugin := false
-		for _, disabledPluginName := range allOffPlugins {
-			if pluginName == disabledPluginName {
-				disablePlugin = true
-				break
-			}
+	enabledPlugins := []string{}
+	allPlugins := []string{}
+	removeDuplicated := sets.NewString()
+
+	// 1.remove duplicated both in PluginNames and DefaultEnabledPlugins
+	for _, plugin := range append(a.PluginNames, a.DefaultEnabledPlugins...) {
+		if !removeDuplicated.Has(plugin) {
+			removeDuplicated.Insert(plugin)
+			allPlugins = append(allPlugins, plugin)
+		}
+	}
```

Comment:
the whole loop can be replaced with `enabledPlugins := sets.Union(sets.NewString(a.PluginNames), sets.NewString(a.DefaultEnabledPlugins))`

Reply:
enabledPlugins := sets.NewString(a.PluginNames).Union(sets.NewString(a.DefaultEnabledPlugins)))

Reply:
No, here I intended to do by iteration, because if `--admission-control=a,b,c,d,e,f,g `, and `RecommendedPluginOrder ` does not contain some of `admission-control`, I want to use specified order.

Reply:
See below, I don't think we should support plugin that are not in the order.

Reply:
Then all the existing plugins should be in the ordered list. We can discuss a reasonable order.

Reply:
The default order is here: https://kubernetes.io/docs/admin/admission-controllers/#is-there-a-recommended-set-of-admission-controllers-to-use